pkg/catalog/util.go

```diff
@@ -319,9 +314,20 @@ func PlantUMLNailGun(contents string) ([]byte, error) {
 	return []byte(plantuml), nil
 }
 
+type MermaidGenerator struct {
+	g *mermaid.Generator
+}
+
+func MakeMermaidGenerator() MermaidGenerator {
+	return MermaidGenerator{g: mermaid.Init()}
+}
+
 // GenerateAndWriteMermaidDiagram writes a mermaid svg to file
-func GenerateAndWriteMermaidDiagram(fs afero.Fs, fileName string, data string) error {
-	mermaidSvg := []byte(mermaid.Execute(data) + "\n")
+func (m MermaidGenerator) GenerateAndWriteMermaidDiagram(fs afero.Fs, fileName string, data string) error {
```

Comment:
Is this fast?

Reply:
Not quite as fast as I'd like it, I think it needs some optimising. Takes around 4s on this repo (and about 50s on `api-catalog`)